### ISSUE TITLE
Use `$templateRequest` instead of `$http` when downloading the template from templateUrl

### ISSUE THREAD

**Issue by matsko** [angular-ui/ui-router#1882]
Would you guys be open to use `$templateRequest` instead of `$http`? I tried creating a patch, but it looks like the tests only run using 1.2. `$templateRequest` runs with 1.3 only. Is there anything we can do? Do you guys have a different test suite for 1.3 code?

I'm trying to fix this bug: https://github.com/angular/angular.js/issues/11492

**Comment by nateabele** [comment]
> Would you guys be open to use `$templateRequest` instead of `$http`?

Yes!

> Do you guys have a different test suite for 1.3 code?

Not currently. We haven't done a version bump of the suite in a while. I'll get on that later today.

**Comment by matsko** [comment]
@nateabele did you get a chance to look into this?

**Comment by nateabele** [comment]
@matsko I did, yeah. Turns out we actually do run the test suite against 1.3, it just wasn't part of the default build (which I have since fixed... I'll add 1.4 shortly as well).

The patch was pretty simple, but it looks like the `$templateFactory` test is hung up in at least two places: differences in the `Accept` header, and what appears to be a BC-break in the API for `$httpBackend`. Open to suggestions on how to resolve that.
